api/src/main/java/org/openmrs/module/dhisconnector/api/impl/DHISConnectorServiceImpl.java

```diff
@@ -1454,6 +1454,26 @@ public String transformToDHISPeriod(Calendar startDate, Calendar endDate, String
 			if (lastRun == null || !(lastRan.get(Calendar.YEAR) + "S" + getHalfYearApril(lastRan))
 					.equals(endDate.get(Calendar.YEAR) + "S" + getHalfYearApril(endDate)))
 				period = startDate.get(Calendar.YEAR) + "S" + getHalfYearApril(startDate);
+		} else if (ReportingPeriodType.Quarterly.name().equals(periodType)) {
+			// Set the correct start date of the last Quarterly period
+			if (startDate.get(Calendar.MONTH) < Calendar.APRIL) {
+				startDate.add(Calendar.YEAR, -1);
+				startDate.set(Calendar.MONTH, Calendar.OCTOBER);
+			} else if (startDate.get(Calendar.MONTH) < Calendar.JULY) {
+				startDate.set(Calendar.MONTH, Calendar.JANUARY);
+			} else if (startDate.get(Calendar.MONTH) < Calendar.OCTOBER) {
+				startDate.set(Calendar.MONTH, Calendar.APRIL);
+			} else {
+				startDate.set(Calendar.MONTH, Calendar.JULY);
+			}
+			startDate.set(Calendar.DAY_OF_MONTH, 1);
+			setBasicsStartsAndEnds(startDate, endDate);
+			// set the end date to the last day of the current period
+			endDate.add(Calendar.MONTH, 2);
+			endDate.set(Calendar.DAY_OF_MONTH, endDate.getActualMaximum(Calendar.DAY_OF_MONTH));
+			if (lastRan == null || !(lastRan.get(Calendar.YEAR) + "Q" + getQuarterNumber(lastRan))
+					.equals(endDate.get(Calendar.YEAR) + "Q" + getQuarterNumber(endDate)))
```

Comment:
don't we need brackets for this if statement ie { }

Reply:
Since it's only one line in the body of the if statement it's not necessary to wrap it around in curly brackets, that's why I didn't put any

Reply:
@HerbertYiga are you still reviewing the PR? 🙂
It'll be great if you can merge this one if there aren't any other issues, cuz we are holding a release for this PR and the next one (which I've already worked on and pushed into different branch in my forked repo) to be merged. 😊

Reply:
I agree with @HerbertYiga,
Maintainability-wise, it's always good to use braces.

Reply:
Alright, Thanks 🤗 
Will do

Reply:
Resolved with the commit [`ec7e220`](https://github.com/openmrs/openmrs-module-dhisconnector/pull/26/commits/ec7e220d07c3dfa458706fffac015c70c76db1fb)
Please review